### ISSUE TITLE
backend naming issue when registering new provider

### ISSUE THREAD

**Issue by ewinston** [QISKit/qiskit-terra#529]
When calling the `register` function which uses a provider which provides backends of the same name as ones previously registered no warning is given.

## Expected Behavior

An exception should be raised if a backend of the same name is already registered. In addition, the register function should provide some way of changing the names of the backends available with the provider being registered e.g. by prefixing/postfixing all backends with a string. 

This could happen through the `backend.__init__(configuration)` call. 
## Current Behavior

It appears that the new provider is added but when getting a backend by name it only gets the first registered.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:
* Environment name and version (e.g. Python 3.6.1):
* Operating System and version:

**Comment by ewinston** [comment]
Some methods of interfacing with new providers,


```
from newprovider import NewProvider
new_provider = NewProvider(token)
qiskit.register(new_provider, name='ibmq')
qiskit.available_backends()
```

Method 2
```
qiskit.register(token, url=URL)
```

Method 3
```
from newprovider import NewProvider
new_provider = NewProvider(*args, **kwargs)
qiskit.register(new_provider)
```

**Comment by ewinston** [comment]
What I was suggesting is that the backend name, perhaps just at the wrapper level, can be composed of a part which the backend defines, e.g. 'qasm_simulator_cpp', plus a name defined for the provider, e.g. 'newpro'. Then in response to `qiskit.available_backends()` you get `newpro_qasm_simulator_cpp`.

**Comment by diego-plan9** [comment]
It seems this was fixed in 0.5.5, in particular at #584  and #602 - please reopen if it is not the case.
